### PR TITLE
Prepare metadata for Nextstrain

### PR DESCRIPTION
**Pull request by ellisrichardj**
This PR makes use of the 'movement.csv' currently generated for `ViewBovine` to extract the appropriate metadata which will be required for building the `.json` files by `Augur` for display in in `Nextstrain`.  This is the first step towards automating the `Nextstrain` build.  `Pandas` has been used for csv manipulation and merging, particularly focussing on consistent use of the Submission number, whilst keeping the Sample number for matching to tree branch labels.  These processes can be updated once the format of data extracted from the metadata warehouse has been finalised.

### CODE REVIEW

**Review by Hira-Naveed — APPROVED** [review]
Richard, this is looking great and making sure we have the appropriate metadata in place for Nextstrain at this stage is super useful and this first step in automation of Nextstrain is a great milestone. I have also used the opportunity for reviewing this PR to learn some Pandas (thank you!).  And thank you for updating on next steps for when we have the MDWH side of things finalised.

File: main.nf [hunk 1]
```diff
@@ -17,6 +17,16 @@ process cleandata {
     """
 }
 
+process metadata {
```
Comment (nick-pestell) [inline main.nf:20]: Would be nice to have a comment describing the process, as on line 30.

File: main.nf [hunk 2]
```diff
@@ -41,6 +51,18 @@ process filterSamples{
     """
 }
 
+process cladeMetadata{
```
Comment (nick-pestell) [inline main.nf:54]: Again, nice to have a comment above. As in line 30.

File: bin/cladeMetadata.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+from datetime import date
+
+def combine(sortedMetadata_csv, cladelist_csv, clade):
+
+    date_out = date.today().strftime('%d%b%y')
+
+    sortedMetadata_df = pd.read_csv(sortedMetadata_csv)
+    cladelist_df = pd.read_csv(cladelist_csv)
+
+    clademetadata_df = pd.merge(cladelist_df, sortedMetadata_df, on = 'Submission', how = 'left')
```
Comment (nick-pestell) [inline bin/cladeMetadata.py:14]: Quite pedantic, but in python keyword arguments, it's standard to have no space before and after the "=".

File: bin/cladeMetadata.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    clademetadata_df.rename(columns = {'Sample': 'name'}, inplace = True)
+    clademetadata_df.set_index('Submission', inplace = True)
+
+    clademetadata_df.to_csv('{}_metadata_{}.csv'.format(clade, date_out))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('sortedMetadata_csv', help = 'path to sortedMetadata.csv')
```
Comment (nick-pestell) [inline bin/cladeMetadata.py:22]: So here as well, ideally there would be no space before and after "=".

File: bin/cladeMetadata.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+
+import pandas as pd
+import argparse
+from datetime import date
+
+def combine(sortedMetadata_csv, cladelist_csv, clade):
```
Comment (nick-pestell) [inline bin/cladeMetadata.py:7]: A docstring would be nice here describing the function, in terms of arguments and return values.

File: bin/filterMetadata.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import pandas as pd
+import argparse
+from datetime import date
+
+def filter(metadata_csv):
+
+    date_out = date.today().strftime('%d%b%y')
+
+    metadata_df = pd.read_csv(metadata_csv)
+    metadata_df['SampleName']=metadata_df['SampleName'].astype(object)
```
Comment (nick-pestell) [inline bin/filterMetadata.py:14]: Here, there should be a space either side of the equals.

File: bin/filterMetadata.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    # sort and deduplicate the metadata
+    metadata_df.sort_values('SampleName', kind='mergesort', inplace = True)
+    metadata_df.sort_values('MovementId', kind='mergesort', inplace = True)
+    metadata_df.drop_duplicates('SampleName', inplace = True)
```
Comment (nick-pestell) [inline bin/filterMetadata.py:18]: I'm not sure what happens on lines 16-18, but I think maybe you're just keeping the first entry for each SampleName?
If that is what's happening, you could perhaps do it without sorting on SampleName first. 
e.g.
```
metadata_df.sort_values("MovementId", kind="mergesort", inplace=True)
metadata_df.drop_duplicates("SampleName", inplace=True)
```

Comment (nick-pestell) [reply bin/filterMetadata.py:18]: Oh sorry, you might also need `keep="first"` as an arg in `drop_duplicates`

**Review by nick-pestell — CHANGES_REQUESTED** [review]
Looks good, just a few things to change

**Comment by ellisrichardj** [comment]
> Looks good, just a few things to change

Thanks - hopefully covered everything here

**Review by nick-pestell — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**